slidingwindow.h

```diff
@@ -1745,4 +1745,235 @@ void ConvolutionInputGenerator_1D_parallel(
 } // End generator
 
 
+/**
+* \brief Sliding Window unit that produces output vectors for feeding
+* a Matrix_Vector_Activate_Batch, implementing the im2col algorithm. To be used for single dimensional feature maps
+*
+* \tparam ConvKernelDim_x    	Dimension of the convolutional kernel - x axis
+* \tparam IFMChannels      	Number of Input Feature Maps
+* \tparam Input_precision  	Number bits per pixel
+* \tparam IFMDim_x          	Width of the Input Feature Map
+* \tparam OFMDim_x           	Width of the Output Feature Map
+* \tparam SIMD             	Number of input columns computed in parallel
+* \tparam Stride_x           	Stride of the convolutional kernel - x axis
+* \tparam R          	  		Datatype for the resource used for FPGA implementation of the SWG  - safely deducible from the parameters
+*
+* \param in                	Input stream
+* \param out               	Output stream
+* \param numReps           	Number of time the function has to be repeatedly executed (e.g. number of images)
+* \param r			  			Resource type for the hardware implementation of the memory block
+*/
+template<unsigned int ConvKernelDim_x,
+	unsigned int IFMChannels,
+	unsigned int Input_precision,
+	unsigned int IFMDim_x,
+	unsigned int OFMDim_x,
+	unsigned int SIMD,
+	unsigned int Stride_x,
+	typename R>
+void ConvolutionInputGenerator_1D(stream<ap_uint<SIMD*Input_precision> > & in, stream<ap_uint<SIMD*Input_precision> > & out, const unsigned int numReps, R const &r) {
+		CASSERT_DATAFLOW(IFMChannels % SIMD == 0);
+
+		const unsigned int multiplying_factor = IFMChannels/SIMD;
+		ap_uint<SIMD*Input_precision> inputBuf[IFMDim_x * multiplying_factor];
+		#pragma HLS ARRAY_PARTITION variable=inputBuf complete dim=1
+		memory_resource(inputBuf, r);
+		const unsigned int cycles_write_block = (OFMDim_x * ConvKernelDim_x * multiplying_factor);
+		const unsigned int cycles_read_block =  IFMDim_x * multiplying_factor;
+		const unsigned int max_cycles = MAX(cycles_write_block,cycles_read_block);
+		const unsigned int baseIter = IFMDim_x * multiplying_factor // Initial buffer
+ 										+ MAX(cycles_write_block,cycles_read_block);
```

Comment:
is this correct? shouldn't the additional term simply be `cycles_write_block`, thus giving `baseIter = cycles_read_block + cycles_write_block` since we have no concurrent reads+writes?